platform/kube/conversion.go

```diff
@@ -82,11 +82,16 @@ func convertPort(port v1.ServicePort) *model.Port {
 
 func convertService(svc v1.Service) *model.Service {
 	ports := make([]*model.Port, 0)
-	addr := ""
+	addr, external := "", false
 	if svc.Spec.ClusterIP != "" && svc.Spec.ClusterIP != v1.ClusterIPNone {
 		addr = svc.Spec.ClusterIP
 	}
 
+	if svc.Spec.Type == v1.ServiceTypeExternalName && svc.Spec.ExternalName != "" {
+		addr = svc.Spec.ExternalName
```

Comment:
I can also use service.Hostname to store the ExternalName value and just not touch service.Address in the external case .  I originally chose to overwrite service.Address with ExternalName because I wanted to preserve the `service.default.local.svc` value because the logic for generating it was not exposed.